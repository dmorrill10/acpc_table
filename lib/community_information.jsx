import React from 'react';
import {CardList, BoardCards, PrivateCards} from './card_list';
import ChipStack from './chip_stack';

class CommunityInformation extends React.Component {
  boardCards() {
    return (<BoardCards key='board-cards' cards={this.props.cards}/>);
  }
  potChips() {
    return <ChipStack key='pot-chips' amount={this.props.potChips}/>;
  }
  render() {
    let l = [this.boardCards(), this.potChips()]
    return <div className='community-information'>{l}</div>;
  }
}
export {CommunityInformation}
