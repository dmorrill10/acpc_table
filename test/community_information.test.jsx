import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import {CommunityInformation} from '../dist/community_information.js';

describe('CommunityInformation', () => {
  describe('without board cards', () => {
    it('works', () => {
      const patient = <CommunityInformation cards={[]} potChips={100}/>;
      const wrapper = shallow(patient);
      expect(renderer.create(patient).toJSON()).toMatchSnapshot();
    });
  });
  describe('with five board cards', () => {
    it('works', () => {
      const cards = [
        {
          rank: '2',
          suit: '&hearts;'
        }, {
          rank: '3',
          suit: '&diamonds;'
        }, {
          rank: '10',
          suit: '&spades;'
        }, {
          rank: 'A',
          suit: '&clubs;'
        }, {
          rank: 'K',
          suit: '&diamonds;'
        }
      ];
      const patient = <CommunityInformation cards={cards} potChips={100}/>;
      const wrapper = shallow(patient);
      expect(renderer.create(patient).toJSON()).toMatchSnapshot();
    });
  });
});
