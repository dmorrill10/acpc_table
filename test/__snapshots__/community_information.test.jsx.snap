exports[`CommunityInformation with five board cards works 1`] = `
<div
  className="community-information">
  <ol
    className="board-cards">
    <li>
      <div
        className="card-face-up">
        <div
          className="top-left-rank">
          2
        </div>
        <div
          className="suit">
          &hearts;
        </div>
        <div
          className="bottom-right-rank">
          2
        </div>
      </div>
    </li>
    <li>
      <div
        className="card-face-up">
        <div
          className="top-left-rank">
          3
        </div>
        <div
          className="suit">
          &diamonds;
        </div>
        <div
          className="bottom-right-rank">
          3
        </div>
      </div>
    </li>
    <li>
      <div
        className="card-face-up">
        <div
          className="top-left-rank">
          10
        </div>
        <div
          className="suit">
          &spades;
        </div>
        <div
          className="bottom-right-rank">
          10
        </div>
      </div>
    </li>
    <li>
      <div
        className="card-face-up">
        <div
          className="top-left-rank">
          A
        </div>
        <div
          className="suit">
          &clubs;
        </div>
        <div
          className="bottom-right-rank">
          A
        </div>
      </div>
    </li>
    <li>
      <div
        className="card-face-up">
        <div
          className="top-left-rank">
          K
        </div>
        <div
          className="suit">
          &diamonds;
        </div>
        <div
          className="bottom-right-rank">
          K
        </div>
      </div>
    </li>
  </ol>
  <div
    className="chip-stack">
    <div
      className="amount">
      100
    </div>
  </div>
</div>
`;

exports[`CommunityInformation without board cards works 1`] = `
<div
  className="community-information">
  <ol
    className="board-cards" />
  <div
    className="chip-stack">
    <div
      className="amount">
      100
    </div>
  </div>
</div>
`;
